src/main/java/net/openhft/hashing/LongTupleHashFunction.java

```diff
@@ -0,0 +1,779 @@
+ * are given for information and could be changed at any moment. However, it could hardly cause
+ * any issues with subclassing, except probably little performance degradation. Methods documented
+ * as "shortcuts" could either delegate to the referenced method or delegate directly to the method
+ * to which the referenced method delegates.
+ *
+ *<p>{@code LongTupleHashFunction} implementations shouldn't assume that {@code Access} strategies
+ * do defensive checks, and access only bytes within the requested range.
+ */
+@ParametersAreNonnullByDefault
+public abstract class LongTupleHashFunction implements Serializable {
+    private static final long serialVersionUID = 0L;
+
+    // Implementations
+    //
+
+    /**
+     * Returns a 128-bit hash function implementing
+     * <a href="https://github.com/aappleby/smhasher/blob/master/src/MurmurHash3.cpp">MurmurHash3
+     * algorithm</a> without seed values. This implementation produces equal results for equal input
+     * on platforms with different {@link ByteOrder}, but is slower on big-endian platforms than on
+     * little-endian.
+     *
+     * @see #murmur_3(long)
+     */
+    @NotNull
+    public static LongTupleHashFunction murmur_3() {
+        return MurmurHash_3.asLongTupleHashFunctionWithoutSeed();
+    }
+
+    /**
+     * Returns a 128-bit hash function implementing
+     * <a href="https://github.com/aappleby/smhasher/blob/master/src/MurmurHash3.cpp">MurmurHash3
+     * algorithm</a> with the given seed value. This implementation produces equal results for equal
+     * input on platforms with different {@link ByteOrder}, but is slower on big-endian platforms
+     * than on little-endian.
+     *
+     * @see #murmur_3()
+     */
+    @NotNull
+    public static LongTupleHashFunction murmur_3(final long seed) {
+        return MurmurHash_3.asLongTupleHashFunctionWithSeed(seed);
+    }
+
+    /**
+     * Constructor for use in subclasses.
+     */
+    protected LongTupleHashFunction() {}
+
+    // Public API
+    //
+
+    /**
+     * Returns the number of bits in a result array; a positive multiple of 8.
+     */
+    public abstract int bitsLength();
+
+    /**
+     * Returns a result array
+     */
+    @NotNull
+    public long[] newResultArray() {
+        return new long[(bitsLength() + 63) / 64];
+    }
+
+    /**
+     * Returns the hash code for the given {@code long} value; this method is consistent with
+     * {@code LongTupleHashFunction} methods that accept sequences of bytes, assuming the {@code input}
+     * value is interpreted in {@linkplain ByteOrder#nativeOrder() native} byte order. For example,
+     * the result of {@code hashLong(v, result)} call is identical to the result of
+     * {@code hashLongs(new long[] {v}, result)} call for any {@code long} value.
+     */
+    public abstract void hashLong(long input, long[] result);
+
+    /**
+     * See {@link #hashLong(long, long[])}
```

Comment:
Since this is a library with an emphasis on zero allocation, I would add a warning note that this method allocates and a suggestion to use another method if the user wants to avoid that.

Reply:
a warning is ok, but we should also explain that:
- in java lang, there is no value types yet, so more than 64-bit info will have to be passed via an object.
- when reusing the `result`, the allocation occurs only at the first time in some runtime scope, after that there is no allocation.